src/Microsoft.AspNet.Routing/Template/TemplateBinder.cs

```diff
@@ -370,11 +370,7 @@ private class TemplateBindingContext
 
                 if (_defaults != null)
                 {
-                    _filters = new RouteValueDictionary(defaults);
-                    foreach (var kvp in _defaults)
-                    {
-                        _filters.Add(kvp.Key, kvp.Value);
-                    }
+                    _filters = new RouteValueDictionary(_defaults);
```

Comment:
- good you fixed the duplicate entries in the `RouteValueDictionary` instance (though that seems unrelated to the point of this PR :smiley_cat: ).  there's still a bug in that `TemplateBindingContext.DebuggerToString()` dereferences `Filters` unconditionally and that property's left `null` if `_defaults == null`. switch to `?.` and all will be well.
- minor but perhaps worth removing the `_filters` and `_acceptedValues` backing fields in `TemplateBindingContext`.  they are only set in that class's constructor; so, a get-only auto-property works.